src/Microsoft.AspNet.Razor.Runtime/DefaultTagHelperContent.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// Default implementation of <see cref="TagHelperContent"/>.
+    /// </summary>
+    public class DefaultTagHelperContent : TagHelperContent, ITextWriterCopyable
+    {
+        private readonly BufferEntryCollection _buffer;
+
+        /// <summary>
+        /// Instantiates a new instance of <see cref="DefaultTagHelperContent"/>.
+        /// </summary>
+        public DefaultTagHelperContent()
+        {
+            _buffer = new BufferEntryCollection();
+        }
+
+        /// <inheritdoc />
+        public override bool IsModified
+        {
+            get
+            {
+                return _buffer.IsModified;
+            }
+        }
+
+        /// <inheritdoc />
+        public override bool IsWhiteSpace
+        {
+            get
+            {
+                foreach (var value in _buffer)
+                {
+                    if (value.Trim().Length != 0)
```

Comment:
@sornaks no harm in a single `null` check here.  this property is also not expected to run through the `string` more than the once `IsNullOrWhiteSpace` would do.